payments-core/src/test/java/com/stripe/android/analytics/DefaultPaymentSessionEventReporterTest.kt

```diff
@@ -95,6 +101,22 @@ class DefaultPaymentSessionEventReporterTest {
         )
     }
 
+    @Test
+    fun `onFormShown() should restart duration on call`() {
+        val durationProvider = mock<DurationProvider> {
+            on { end(any()) } doReturn Duration.ZERO
+        }
```

Comment:
You can use `FakeDurationProvider`.

Reply:
Can we make that possible in the fake? For instance, expose the “active” keys as a property that can be asserted against? I just want us to move away from mocks for objects that we control.